src/ext/oblivc/obliv_transport.c

```diff
@@ -0,0 +1,343 @@
+#include <obliv_common.h>
+#include <obliv_bits.h>
+#include <obliv_yao.h>
+#include <commitReveal.h>
+#include <nnob.h>
+#include <assert.h>
+#include <errno.h>      // libgcrypt needs ENOMEM definition
+#include <inttypes.h>
+#include <stdio.h>      // for protoUseStdio()
+#include <string.h>
+#include <netinet/ip.h>
+#include <netinet/tcp.h>
+#include <netdb.h>
+#include <pthread.h>
+#include <unistd.h>
+#include <gcrypt.h>
+
+// --------------------------- Transports -----------------------------------
+struct stdioTransport
+{ ProtocolTransport cb;
+  bool needFlush;
+};
+
+// Ignores 'dest' parameter. So you can't osend to yourself
+static bool* stdioFlushFlag(ProtocolTransport* pt) 
+  { return &((struct stdioTransport*)pt)->needFlush; }
+
+static int stdioSend(ProtocolTransport* pt,int dest,const void* s,size_t n)
+{ *stdioFlushFlag(pt)=true;
+  return fwrite(s,1,n,stdout); 
+}
+
+static int stdioRecv(ProtocolTransport* pt,int src,void* s,size_t n)
+{ 
+  bool *p = stdioFlushFlag(pt);
+  if(*p) { fflush(stdout); *p=false; }
+  return fread(s,1,n,stdin); 
+}
+
+static void stdioCleanup(ProtocolTransport* pt) {}
+
+// Extremely simple, no multiplexing: two parties, one connection
+static struct stdioTransport stdioTransport 
+  = {{2, NULL, stdioSend, stdioRecv, stdioCleanup},false};
+
+void protocolUseStdio(ProtocolDesc* pd)
+  { pd->trans = &stdioTransport.cb; }
+
```

Comment:
I'm not sure what happen here...could it be that I had starting making my changes on a branch that did not have your changes? And now I'm merging back to master so it might not appear, or actually be, overwritten? Could you point me to where I can see this overwritten code? Thanks, and sorry about that!